app/artifact-cas/internal/server/grpc.go

```diff
@@ -93,6 +94,18 @@ func NewGRPCServer(c *conf.Server, authConf *conf.Auth, byteService *service.Byt
 	if c.Grpc.Timeout != nil {
 		opts = append(opts, grpc.Timeout(c.Grpc.Timeout.AsDuration()))
 	}
+	cert := c.Grpc.TlsConfig.Certificate
+	privKey := c.Grpc.TlsConfig.PrivateKey
+	if cert != "" && privKey != "" {
+		cert, err := tls.LoadX509KeyPair(cert, privKey)
+		if err != nil {
+			return nil, err
```

Comment:
it might make sense to return wrapped context in the error i.e `fmt.Errorf("loading gRPC server TLS certificate: %w", err)`